client/session.go

```diff
@@ -407,13 +426,27 @@ func (s *session) Open() error {
 		))
 	s.writeOperationPool = newWriteOperationPool(writeOperationPoolOpts)
 	s.writeOperationPool.Init()
+
+	writeTaggedOperationPoolOpts := pool.NewObjectPoolOptions().
+		SetSize(s.opts.WriteTaggedOpPoolSize()).
+		SetInstrumentOptions(s.opts.InstrumentOptions().SetMetricsScope(
+			s.scope.SubScope("write-op-tagged-pool"),
+		))
+	s.writeTaggedOperationPool = newWriteTaggedOpPool(writeTaggedOperationPoolOpts)
+	s.writeTaggedOperationPool.Init()
+
+	writeStatePoolSize := s.opts.WriteOpPoolSize()
```

Comment:
also have to do these checks because of the two kinds of pools using the writestate